src/common/BaseComponent.ts

```diff
@@ -12,7 +12,7 @@ export class BaseComponent<P, S> extends React.Component<P, S> {
 
   private __async: Async;
   private __events: EventGroup;
-  private __disposables: IDisposable[];
+  private __disposables: (IDisposable | (() => void))[];
```

Comment:
I think there are cases for both but I am not sure it's necessary to enforce one pattern over the other. Why would you desire to enforce a totally unnecessary object wrapper around a simple callback when it's very simple to support either?